src/profile-logic/profile-data.js

```diff
@@ -2577,12 +2577,20 @@ export function extractProfileFilterPageData(
     return null;
   }
 
-  // Getting the first page's innerWindowID and then getting its url.
-  const innerWindowID = [...relevantPages][0];
-  const filteredPages = pages.filter(
-    page => page.innerWindowID === innerWindowID
+  // Getting the pages that are relevant and a topmost frame.
+  let filteredPages = pages.filter(
+    page =>
+      relevantPages.has(page.innerWindowID) && page.embedderInnerWindowID === 0
   );
 
+  if (filteredPages.length > 1) {
+    // If there are more than one topmost page, it's also good to filter out the
+    // `about:` pages so user can see their url they are actually profiling.
+    filteredPages = filteredPages.filter(
+      page => !page.url.startsWith('about:')
+    );
+  }
+
   if (filteredPages.length !== 1) {
     // There should be only one page with the given innerWindowID, they are unique.
     console.error(`Expected one page but ${filteredPages.length} found.`);
```

Comment:
Can we now have cases where filteredPates.length is > 1, now that we check all windowIds in `relevantPages`? Is it guaranteed we'll find only one? Maybe we should we take the first one in that case, instead of returning null?
For example can we navigate from eg `google.com` to `cnn.com`, and get 2 pages, and so get an error here?

Can you please explicit this?

Note that the comment at the top of the function is:
```
/**
 * Extract the hostname and favicon from the first page if we are in single tab
 * view. Currently we assume that we don't change the origin of webpages while
 * profiling in web developer preset. That's why we are simply getting the first
 * page we find that belongs to the active tab. Returns null if profiler is not
 * in the single tab view at the moment.
 */
```
so this would make sense to get the first `filteredPages` if we have more than one.